meddlr/data/transforms/transform.py

```diff
@@ -364,3 +365,211 @@ def __call__(self, kspace, maps, target, fname, slice_id, is_fixed, acceleration
+            is_fixed (bool, optional): If `True`, transform the example
+                to have a fixed mask and acceleration factor.
+            acceleration (int): Acceleration factor. Must be provided if
+                `is_undersampled=True`.
+        Returns:
+            (tuple): tuple containing:
+                image (torch.Tensor): Zero-filled input image.
+                target (torch.Tensor): Target image converted to a torch Tensor.
+                mean (float): Mean value used for normalization.
+                std (float): Standard deviation value used for normalization.
+                norm (float): L2 norm of the entire volume.
+        """
+        if is_fixed and not acceleration:
+            raise ValueError("Accelerations must be specified for undersampled scans")
+
+        # If augmentor is defined, use it to do computation.
+        if self.augmentor is not None:
+            return self._call_augmentor(
+                kspace, maps, target, fname, slice_id, is_fixed, acceleration
+            )
+
+        # Convert everything from numpy arrays to tensors
+        kspace = cplx.to_tensor(kspace).unsqueeze(0)
+        maps = cplx.to_tensor(maps).unsqueeze(0)
+        target_init = cplx.to_tensor(target).unsqueeze(0)
+        target = (
+            torch.complex(target_init, torch.zeros_like(target_init)).unsqueeze(-1)
+            if not torch.is_complex(target_init)
+            else target_init
+        )  # handle rss vs. sensitivity-integrated
+        norm = torch.sqrt(torch.mean(cplx.abs(target) ** 2))
+
+        # TODO: Add other transforms here.
+
+        # Apply mask in k-space
+        seed = sum(tuple(map(ord, fname))) if self._is_test or is_fixed else None  # noqa
+        masked_kspace, mask = self._subsampler(
+            kspace, mode="2D", seed=seed, acceleration=acceleration
+        )
+
+        # Zero-filled Sense Recon.
+        if torch.is_complex(target_init):
+            A = SenseModel(maps, weights=mask)
+            image = A(masked_kspace, adjoint=True)
+        # Zero-filled RSS Recon.
+        else:
+            image = T.ifft2(masked_kspace)
+            image_rss = torch.sqrt(torch.sum(cplx.abs(image) ** 2, axis=-1))
+            image = torch.complex(image_rss, torch.zeros_like(image_rss)).unsqueeze(-1)
+
+        # Normalize
+        normalized = self._normalizer.normalize(
+            **{"masked_kspace": masked_kspace, "image": image, "target": target, "mask": mask}
+        )
+        masked_kspace = normalized["masked_kspace"]
+        target = normalized["target"]
+        mean = normalized["mean"]
+        std = normalized["std"]
+
+        add_noise = self.add_noise and (
+            self._is_test or (not is_fixed and self.rng.uniform() < self.p_noise)
+        )
+        add_motion = self.add_motion and (
+            self._is_test or (not is_fixed and self.rng.uniform() < self.p_motion)
+        )
+        if add_motion:
```

Comment:
motion simulation should happen before the undersampling

Reply:
Where should the noise simulation go? Before or after the undersampling? Thanks, Deepro.